README.md

```diff
@@ -240,22 +240,527 @@ For complete documentation of the `mp-build` tool see the [MP Build Tool Documen
+    credential.define_password_parameter("password", "Password")
+
+    # The key 'container_memory_limit' is a special key that is read by the VMware Aria 
+    # Operations collector to determine how much memory to allocate to the docker 
+    # container running this adapter. It does not need to be read inside the adapter 
+    # code.
+    definition.define_int_parameter(
+        "container_memory_limit",
+        label="Adapter Memory Limit (MB)",
+        description="Sets the maximum amount of memory VMware Aria Operations can "
+                    "allocate to the container running this adapter instance.",
+        required=True,
+        advanced=True,
+        default=1024,
+    )
+
+    # This Adapter has no object types directly, rather it co-opts object types that
+    # are part of the MySQL MP to add additional metrics. As such, we can't define
+    # any object types here, because they're already defined in the MySQL MP with a
+    # different adapter type.
+
+    logger.info("Finished 'Get Adapter Definition'")
+    logger.debug(f"Returning adapter definition: {definition.to_json()}")
+    return definition
+```
+
+The adapter definition is also where objects and metrics are defined, however we are 
+only allowed to define objects and metrics that are a part of our adapter type. Because
+extensions modify objects that are part of a different adapter type, we can't add them.
+This means that we can't set metric metadata like 'units', 'labels', etc that we would
+generally be able to set.
+
+#### Modify the `test` method to create a MySQL connection and run a query
+
+Now that we've defined our parameters, we can try to connect and run a test query.
+We will do this in the `test` method. Notice this takes an `AdapterInstance` as a 
+parameter. All the parameters from the definition will be present in this Adapter 
+Instance. We can access them like this:
+
+```python
+    hostname = adapter_instance.get_identifier_value("host")
+    port = int(adapter_instance.get_identifier_value("port", "3306"))
+    username = adapter_instance.get_credential_value("username")
+    password = adapter_instance.get_credential_value("password")
+```
+
+We can then use them to connect to MySQL and run a test query:
+
+```python
+    connection = mysql.connector.connect(
+        host=hostname,
+        port=port,
+        user=username,
+        password=password,
+    )
+    cursor = connection.cursor()
+
+    # Run a simple test query
+    cursor.execute("SHOW databases")
+    cursor.close()
+```
+
+Since we can expect that this will fail, e.g., if the user provides the wrong username
+and password, we should ensure there is good error-handling in this function. If the 
+management pack will be widely distributed, it may also be worthwhile to catch common 
+errors and ensure the resulting messages are user-friendly.
+
+We should now be able to run `mp-test connect` to run this code. The `mp-test` tool
```

Comment:
In addition to clarifying where the new code goes and what needs to be removed, we should clarify that the `TestResult()` returning without modification indicates success.